syntax = "proto3";

// option objc_class_prefix = "HLW";

package ai_server;

service Segmentation {
  rpc Start (RequestStart) returns (ReplyRequestStart) {}
  rpc UpdateMask(RequestMaskUpdate) returns (ReplyMaskUpdate) {}
  rpc OnComplete (SendResult) returns (ReplySendResult) {}
}

message RequestMaskUpdate{
  int64 req_id = 1;
  bytes mask = 2;
}

message ReplyMaskUpdate{
  int64 req_id = 1;
  bytes mask = 2;
  bytes inpaint = 3;
}

message SendResult {
  int64 req_id = 1;
  string data = 2;
}

message ReplySendResult {
  int64 req_id = 1;
  int32 status_code = 2;
}

message RequestStart {
  int64 req_id = 1;
  bytes image = 2;
}

message ReplyRequestStart {
  int64 req_id = 1;
  int32 status_code = 2;
}